imap/auth_oauth.c

```diff
@@ -79,15 +79,15 @@ enum ImapAuthRes imap_auth_oauth(struct ImapAccountData *adata, const char *meth
   FREE(&oauthbearer);
   FREE(&ibuf);
 
-  if (rc == IMAP_EXEC_SUCCESS)
+  if (rc != IMAP_EXEC_SUCCESS)
   {
     /* The error response was in SASL continuation, so continue the SASL
      * to cause a failure and exit SASL input.  See RFC 7628 3.2.3 */
     mutt_socket_send(adata->conn, "\001");
     rc = imap_exec(adata, ibuf, IMAP_CMD_NO_FLAGS);
   }
 
-  if (rc != IMAP_EXEC_SUCCESS)
+  if (rc == IMAP_EXEC_SUCCESS)
```

Comment:
is it worth making this an `else` ?

Reply:
rc can be the one from L77 or L87, so we can't use else.

Reply:
Sorry, don't mind me -- I'm a liability :-)